The job **`aarch64-gnu-llvm-19-1`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/16857881103/job/47753741271) [(plain enhanced)](https://triage.rust-lang.org/gha-logs/rust-lang/rust/47753741271) [(plain)](https://github.com/rust-lang/rust/commit/7bd902dcf5800f8179c74405bf15869721950cc7/checks/47753741271/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Saved the actual stderr to `/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/attributes/malformed-attrs/malformed-attrs.stderr`
diff of stderr:

662 For more information about an error, try `rustc --explain E0308`.
663 Future incompatibility report: Future breakage diagnostic:
664 error: valid forms for the attribute are `#[doc(hidden|inline|...)]` and `#[doc = "string"]`
-   --> $DIR/malformed-attrs.rs:44:1
+   --> $DIR/malformed-attrs.rs:40:1
666    |
667 LL | #[doc]
668    | ^^^^^^

673 
674 Future breakage diagnostic:
675 error: valid forms for the attribute are `#[doc(hidden|inline|...)]` and `#[doc = "string"]`
-   --> $DIR/malformed-attrs.rs:77:1
+   --> $DIR/malformed-attrs.rs:73:1
677    |
678 LL | #[doc]
679    | ^^^^^^

684 
685 Future breakage diagnostic:
686 error: attribute must be of the form `#[link(name = "...", /*opt*/ kind = "dylib|static|...", /*opt*/ wasm_import_module = "...", /*opt*/ import_name_type = "decorated|noprefix|undecorated")]`
-   --> $DIR/malformed-attrs.rs:84:1
+   --> $DIR/malformed-attrs.rs:80:1
688    |
689 LL | #[link]
690    | ^^^^^^^

695 
696 Future breakage diagnostic:
697 error: valid forms for the attribute are `#[inline(always|never)]` and `#[inline]`
-   --> $DIR/malformed-attrs.rs:54:1
+   --> $DIR/malformed-attrs.rs:50:1
699    |
700 LL | #[inline = 5]
701    | ^^^^^^^^^^^^^

706 
707 Future breakage diagnostic:
708 error: valid forms for the attribute are `#[ignore = "reason"]` and `#[ignore]`
-   --> $DIR/malformed-attrs.rs:95:1
+   --> $DIR/malformed-attrs.rs:91:1
710    |
711 LL | #[ignore()]
712    | ^^^^^^^^^^^

717 
718 Future breakage diagnostic:
719 error: valid forms for the attribute are `#[ignore = "reason"]` and `#[ignore]`
-   --> $DIR/malformed-attrs.rs:224:1
+   --> $DIR/malformed-attrs.rs:220:1
721    |
722 LL | #[ignore = 1]
723    | ^^^^^^^^^^^^^
---
To only update this specific test, also pass `--test-args attributes/malformed-attrs.rs`

error: 1 errors occurred comparing output.
status: exit status: 1
command: env -u RUSTC_LOG_COLOR RUSTC_ICE="0" RUST_BACKTRACE="short" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/tests/ui/attributes/malformed-attrs.rs" "-Zthreads=1" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Z" "ignore-directory-in-diagnostics-source-blocks=/cargo" "-Z" "ignore-directory-in-diagnostics-source-blocks=/checkout/vendor" "--sysroot" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2" "--target=aarch64-unknown-linux-gnu" "--check-cfg" "cfg(test,FALSE)" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zwrite-long-types-to-disk=no" "-Cstrip=debuginfo" "--emit" "metadata" "-C" "prefer-dynamic" "--out-dir" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/attributes/malformed-attrs" "-A" "unused" "-A" "internal_features" "-A" "unused_parens" "-A" "unused_braces" "-Crpath" "-Cdebuginfo=0" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers"
stdout: none
--- stderr -------------------------------
error[E0539]: malformed `cfg` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:99:1
   |
LL | #[cfg]
   | ^^^^^^
   | |
   | expected this to be a list
   | help: must be of the form: `#[cfg(predicate)]`

error: malformed `cfg_attr` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:101:1
   |
LL | #[cfg_attr]
---

error: malformed `windows_subsystem` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:26:1
   |
LL | #![windows_subsystem]
   | ^^^^^^^^^^^^^^^^^^^^^ help: must be of the form: `#![windows_subsystem = "windows|console"]`

error: malformed `crate_name` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:71:1
   |
LL | #[crate_name]
   | ^^^^^^^^^^^^^ help: must be of the form: `#[crate_name = "name"]`

error: malformed `no_sanitize` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:89:1
   |
LL | #[no_sanitize]
   | ^^^^^^^^^^^^^^ help: must be of the form: `#[no_sanitize(address, kcfi, memory, thread)]`

error: malformed `instruction_set` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:103:1
   |
LL | #[instruction_set]
   | ^^^^^^^^^^^^^^^^^^ help: must be of the form: `#[instruction_set(set)]`

error: malformed `patchable_function_entry` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:105:1
   |
LL | #[patchable_function_entry]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: must be of the form: `#[patchable_function_entry(prefix_nops = m, entry_nops = n)]`

error: malformed `must_not_suspend` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:129:1
   |
LL | #[must_not_suspend()]
   | ^^^^^^^^^^^^^^^^^^^^^
   |
help: the following are the possible correct uses
   |
LL - #[must_not_suspend()]
LL + #[must_not_suspend = "reason"]
   |
LL - #[must_not_suspend()]
LL + #[must_not_suspend]
   |

error: malformed `cfi_encoding` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:131:1
   |
LL | #[cfi_encoding]
   | ^^^^^^^^^^^^^^^ help: must be of the form: `#[cfi_encoding = "encoding"]`

error: malformed `linkage` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:170:5
   |
LL |     #[linkage]
   |     ^^^^^^^^^^ help: must be of the form: `#[linkage = "external|internal|..."]`

error: malformed `allow` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:175:1
   |
LL | #[allow]
   | ^^^^^^^^ help: must be of the form: `#[allow(lint1, lint2, ..., /*opt*/ reason = "...")]`

error: malformed `expect` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:177:1
   |
LL | #[expect]
   | ^^^^^^^^^ help: must be of the form: `#[expect(lint1, lint2, ..., /*opt*/ reason = "...")]`

error: malformed `warn` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:179:1
   |
LL | #[warn]
   | ^^^^^^^ help: must be of the form: `#[warn(lint1, lint2, ..., /*opt*/ reason = "...")]`

error: malformed `deny` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:181:1
   |
LL | #[deny]
   | ^^^^^^^ help: must be of the form: `#[deny(lint1, lint2, ..., /*opt*/ reason = "...")]`

error: malformed `forbid` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:183:1
   |
LL | #[forbid]
   | ^^^^^^^^^ help: must be of the form: `#[forbid(lint1, lint2, ..., /*opt*/ reason = "...")]`

error: malformed `debugger_visualizer` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:185:1
   |
LL | #[debugger_visualizer]
   | ^^^^^^^^^^^^^^^^^^^^^^ help: must be of the form: `#[debugger_visualizer(natvis_file = "...", gdb_script_file = "...")]`

error: malformed `thread_local` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:200:1
   |
LL | #[thread_local()]
   | ^^^^^^^^^^^^^^^^^ help: must be of the form: `#[thread_local]`

error: malformed `no_link` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:204:1
   |
LL | #[no_link()]
   | ^^^^^^^^^^^^ help: must be of the form: `#[no_link]`

error: malformed `macro_export` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:211:1
   |
LL | #[macro_export = 18]
---
   |
LL | #[allow_internal_unsafe = 1]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: must be of the form: `#[allow_internal_unsafe]`

error: the `#[proc_macro]` attribute is only usable with crates of the `proc-macro` crate type
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:96:1
   |
LL | #[proc_macro = 18]
   | ^^^^^^^^^^^^^^^^^^

error: the `#[proc_macro_attribute]` attribute is only usable with crates of the `proc-macro` crate type
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:113:1
   |
LL | #[proc_macro_attribute = 19]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: the `#[proc_macro_derive]` attribute is only usable with crates of the `proc-macro` crate type
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:120:1
   |
LL | #[proc_macro_derive]
   | ^^^^^^^^^^^^^^^^^^^^

error[E0658]: allow_internal_unsafe side-steps the unsafe_code lint
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:213:1
   |
LL | #[allow_internal_unsafe = 1]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = help: add `#![feature(allow_internal_unsafe)]` to the crate attributes to enable
   = note: this compiler was built on YYYY-MM-DD; consider upgrading it if it is out of date

error: valid forms for the attribute are `#[doc(hidden|inline|...)]` and `#[doc = "string"]`
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:40:1
   |
LL | #[doc]
   | ^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #57571 <https://github.com/rust-lang/rust/issues/57571>
   = note: `#[deny(ill_formed_attribute_input)]` on by default

error: valid forms for the attribute are `#[doc(hidden|inline|...)]` and `#[doc = "string"]`
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:73:1
   |
LL | #[doc]
   | ^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #57571 <https://github.com/rust-lang/rust/issues/57571>

error: attribute must be of the form `#[link(name = "...", /*opt*/ kind = "dylib|static|...", /*opt*/ wasm_import_module = "...", /*opt*/ import_name_type = "decorated|noprefix|undecorated")]`
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:80:1
   |
LL | #[link]
   | ^^^^^^^
   |
---

error[E0539]: malformed `export_name` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:29:1
   |
LL | #[unsafe(export_name)]
   | ^^^^^^^^^^^^^^^^^^^^^^ help: must be of the form: `#[export_name = "name"]`

error: `rustc_allow_const_fn_unstable` expects a list of feature names
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:31:1
   |
LL | #[rustc_allow_const_fn_unstable]
---
LL | #[rustc_confusables]
   | ^^^^^^^^^^^^^^^^^^^^
   | |
   | expected this to be a list
   | help: must be of the form: `#[rustc_confusables("name1", "name2", ...)]`

error[E0539]: malformed `deprecated` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:38:1
   |
LL | #[deprecated = 5]
   | ^^^^^^^^^^^^^^^-^
   |                |
   |                expected a string literal here
   |
help: try changing it to one of the following valid forms of the attribute
   |
LL - #[deprecated = 5]
LL + #[deprecated = "reason"]
   |
LL - #[deprecated = 5]
LL + #[deprecated(/*opt*/ since = "version", /*opt*/ note = "reason")]
   |
LL - #[deprecated = 5]
LL + #[deprecated]
   |

error[E0539]: malformed `rustc_macro_transparency` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:43:1
   |
LL | #[rustc_macro_transparency]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^ help: must be of the form: `#[rustc_macro_transparency = "transparent|semitransparent|opaque"]`

error[E0539]: malformed `repr` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:45:1
   |
LL | #[repr]
   | ^^^^^^^
   | |
   | expected this to be a list
   | help: must be of the form: `#[repr(C | Rust | align(...) | packed(...) | <integer type> | transparent)]`

error[E0565]: malformed `rustc_as_ptr` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:48:1
   |
LL | #[rustc_as_ptr = 5]
   | ^^^^^^^^^^^^^^^---^
   | |              |
   | |              didn't expect any arguments here
   | help: must be of the form: `#[rustc_as_ptr]`

error[E0539]: malformed `rustc_align` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:53:1
   |
LL | #[rustc_align]
   | ^^^^^^^^^^^^^^
   | |
   | expected this to be a list
   | help: must be of the form: `#[rustc_align(<alignment in bytes>)]`

error[E0539]: malformed `optimize` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:55:1
   |
LL | #[optimize]
   | ^^^^^^^^^^^
   | |
   | expected this to be a list
   | help: must be of the form: `#[optimize(size|speed|none)]`

error[E0565]: malformed `cold` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:57:1
   |
LL | #[cold = 1]
   | ^^^^^^^---^
   | |      |
   | |      didn't expect any arguments here
   | help: must be of the form: `#[cold]`

error: valid forms for the attribute are `#[must_use = "reason"]` and `#[must_use]`
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:59:1
   |
LL | #[must_use()]
   | ^^^^^^^^^^^^^

---

error[E0539]: malformed `export_name` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:67:1
   |
LL | #[export_name()]
   | ^^^^^^^^^^^^^^^^ help: must be of the form: `#[export_name = "name"]`

error[E0805]: malformed `used` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:69:1
   |
LL | #[used()]
   | ^^^^^^--^
   |       |
   |       expected a single argument here
   |
help: try changing it to one of the following valid forms of the attribute
   |
LL | #[used(compiler|linker)]
   |        +++++++++++++++
LL - #[used()]
LL + #[used]
   |

error[E0539]: malformed `target_feature` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:76:1
   |
LL | #[target_feature]
   | ^^^^^^^^^^^^^^^^^
   | |
   | expected this to be a list
   | help: must be of the form: `#[target_feature(enable = "feat1, feat2")]`

error[E0565]: malformed `export_stable` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:78:1
   |
LL | #[export_stable = 1]
---
error[E0539]: malformed `link_name` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:83:1
   |
LL | #[link_name]
   | ^^^^^^^^^^^^ help: must be of the form: `#[link_name = "name"]`

error[E0539]: malformed `link_section` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:85:1
   |
LL | #[link_section]
   | ^^^^^^^^^^^^^^^ help: must be of the form: `#[link_section = "name"]`

error[E0539]: malformed `coverage` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:87:1
   |
LL | #[coverage]
   | ^^^^^^^^^^^ this attribute is only valid with either `on` or `off` as an argument
   |
help: try changing it to one of the following valid forms of the attribute
   |
LL | #[coverage(off)]
   |           +++++
---
   |
LL | #[no_implicit_prelude = 23]
   | ^^^^^^^^^^^^^^^^^^^^^^----^
   | |                     |
   | |                     didn't expect any arguments here
   | help: must be of the form: `#[no_implicit_prelude]`

error[E0565]: malformed `proc_macro` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:96:1
   |
LL | #[proc_macro = 18]
   | ^^^^^^^^^^^^^----^
   | |            |
   | |            didn't expect any arguments here
   | help: must be of the form: `#[proc_macro]`

error[E0565]: malformed `coroutine` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:108:5
   |
LL |     #[coroutine = 63] || {}
---
   |
LL | #[proc_macro_attribute = 19]
   | ^^^^^^^^^^^^^^^^^^^^^^^----^
   | |                      |
   | |                      didn't expect any arguments here
   | help: must be of the form: `#[proc_macro_attribute]`

error[E0539]: malformed `must_use` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:116:1
   |
LL | #[must_use = 1]
   | ^^^^^^^^^^^^^-^
   |              |
   |              expected a string literal here
   |
help: try changing it to one of the following valid forms of the attribute
   |
LL - #[must_use = 1]
LL + #[must_use = "reason"]
   |
LL - #[must_use = 1]
LL + #[must_use]
   |

error[E0539]: malformed `proc_macro_derive` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:120:1
   |
LL | #[proc_macro_derive]
   | ^^^^^^^^^^^^^^^^^^^^
   | |
   | expected this to be a list
   | help: must be of the form: `#[proc_macro_derive(TraitName, /*opt*/ attributes(name1, name2, ...))]`

error[E0539]: malformed `rustc_layout_scalar_valid_range_start` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:125:1
   |
LL | #[rustc_layout_scalar_valid_range_start]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   | |
   | expected this to be a list
   | help: must be of the form: `#[rustc_layout_scalar_valid_range_start(start)]`

error[E0539]: malformed `rustc_layout_scalar_valid_range_end` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:127:1
   |
LL | #[rustc_layout_scalar_valid_range_end]
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   | |
   | expected this to be a list
   | help: must be of the form: `#[rustc_layout_scalar_valid_range_end(end)]`

error[E0565]: malformed `marker` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:152:1
   |
LL | #[marker = 3]
   | ^^^^^^^^^---^
   | |        |
   | |        didn't expect any arguments here
   | help: must be of the form: `#[marker]`

error[E0565]: malformed `fundamental` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:154:1
   |
LL | #[fundamental()]
   | ^^^^^^^^^^^^^--^
   | |            |
   | |            didn't expect any arguments here
   | help: must be of the form: `#[fundamental]`

error[E0565]: malformed `ffi_pure` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:162:5
   |
LL |     #[unsafe(ffi_pure = 1)]
   |     ^^^^^^^^^^^^^^^^^^---^^
   |     |                 |
   |     |                 didn't expect any arguments here
   |     help: must be of the form: `#[ffi_pure]`

error[E0539]: malformed `link_ordinal` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:164:5
   |
LL |     #[link_ordinal]
   |     ^^^^^^^^^^^^^^^
   |     |
   |     expected this to be a list
   |     help: must be of the form: `#[link_ordinal(ordinal)]`

error[E0565]: malformed `ffi_const` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:168:5
   |
LL |     #[unsafe(ffi_const = 1)]
   |     ^^^^^^^^^^^^^^^^^^^---^^
   |     |                  |
   |     |                  didn't expect any arguments here
   |     help: must be of the form: `#[ffi_const]`

error[E0565]: malformed `automatically_derived` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:188:1
   |
LL | #[automatically_derived = 18]
---
   | |                |
   | |                didn't expect any arguments here
   | help: must be of the form: `#[non_exhaustive]`

error: valid forms for the attribute are `#[macro_use(name1, name2, ...)]` and `#[macro_use]`
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:206:1
   |
LL | #[macro_use = 1]
   | ^^^^^^^^^^^^^^^^

error[E0565]: malformed `type_const` attribute input
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:140:5
   |
LL |     #[type_const = 1]
   |     ^^^^^^^^^^^^^---^
   |     |            |
   |     |            didn't expect any arguments here
   |     help: must be of the form: `#[type_const]`

error: attribute should be applied to `const fn`
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:31:1
   |
LL |   #[rustc_allow_const_fn_unstable]
---
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ invalid option found here
   |
   = help: only `message`, `note` and `label` are allowed as options

error: valid forms for the attribute are `#[inline(always|never)]` and `#[inline]`
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:50:1
   |
LL | #[inline = 5]
   | ^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #57571 <https://github.com/rust-lang/rust/issues/57571>

error: valid forms for the attribute are `#[ignore = "reason"]` and `#[ignore]`
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:91:1
   |
LL | #[ignore()]
   | ^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #57571 <https://github.com/rust-lang/rust/issues/57571>

error: valid forms for the attribute are `#[ignore = "reason"]` and `#[ignore]`
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:220:1
   |
LL | #[ignore = 1]
   | ^^^^^^^^^^^^^
   |
---
LL |     #[coroutine = 63] || {}
   |                       ^^^^^ expected `()`, found coroutine
   |
   = note: expected unit type `()`
              found coroutine `{coroutine@/checkout/tests/ui/attributes/malformed-attrs.rs:108:23: 108:25}`

error: aborting due to 74 previous errors; 3 warnings emitted

Some errors have detailed explanations: E0308, E0463, E0539, E0565, E0658, E0805.
For more information about an error, try `rustc --explain E0308`.
Future incompatibility report: Future breakage diagnostic:
error: valid forms for the attribute are `#[doc(hidden|inline|...)]` and `#[doc = "string"]`
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:40:1
   |
LL | #[doc]
   | ^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #57571 <https://github.com/rust-lang/rust/issues/57571>
   = note: `#[deny(ill_formed_attribute_input)]` on by default

Future breakage diagnostic:
error: valid forms for the attribute are `#[doc(hidden|inline|...)]` and `#[doc = "string"]`
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:73:1
   |
LL | #[doc]
   | ^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #57571 <https://github.com/rust-lang/rust/issues/57571>
   = note: `#[deny(ill_formed_attribute_input)]` on by default

Future breakage diagnostic:
error: attribute must be of the form `#[link(name = "...", /*opt*/ kind = "dylib|static|...", /*opt*/ wasm_import_module = "...", /*opt*/ import_name_type = "decorated|noprefix|undecorated")]`
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:80:1
   |
LL | #[link]
   | ^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #57571 <https://github.com/rust-lang/rust/issues/57571>
   = note: `#[deny(ill_formed_attribute_input)]` on by default

Future breakage diagnostic:
error: valid forms for the attribute are `#[inline(always|never)]` and `#[inline]`
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:50:1
   |
LL | #[inline = 5]
   | ^^^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #57571 <https://github.com/rust-lang/rust/issues/57571>
   = note: `#[deny(ill_formed_attribute_input)]` on by default

Future breakage diagnostic:
error: valid forms for the attribute are `#[ignore = "reason"]` and `#[ignore]`
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:91:1
   |
LL | #[ignore()]
   | ^^^^^^^^^^^
   |
   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
   = note: for more information, see issue #57571 <https://github.com/rust-lang/rust/issues/57571>
   = note: `#[deny(ill_formed_attribute_input)]` on by default

Future breakage diagnostic:
error: valid forms for the attribute are `#[ignore = "reason"]` and `#[ignore]`
##[error]  --> /checkout/tests/ui/attributes/malformed-attrs.rs:220:1
   |
LL | #[ignore = 1]
   | ^^^^^^^^^^^^^
   |
```

</details>

--- a/compiler/rustc_lint_defs/src/builtin.rs
+++ b/compiler/rustc_lint_defs/src/builtin.rs
@@ -217,6 +217,7 @@ declare_lint! {
     @future_incompatible = FutureIncompatibleInfo {
         reason: FutureIncompatibilityReason::FutureReleaseError,
         reference: "issue #57571 <https://github.com/rust-lang/rust/issues/57571>",
+        report_in_deps: true,
     };
     crate_level_only
 }

--- a/tests/ui/attributes/lint_on_root.stderr
+++ b/tests/ui/attributes/lint_on_root.stderr
@@ -10,3 +10,14 @@ LL | #![inline = ""]
 
 error: aborting due to 1 previous error
 
+Future incompatibility report: Future breakage diagnostic:
+error: valid forms for the attribute are `#[inline(always|never)]` and `#[inline]`
+  --> $DIR/lint_on_root.rs:3:1
+   |
+LL | #![inline = ""]
+   | ^^^^^^^^^^^^^^^
+   |
+   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
+   = note: for more information, see issue #57571 <https://github.com/rust-lang/rust/issues/57571>
+   = note: `#[deny(ill_formed_attribute_input)]` on by default
+

--- a/tests/ui/attributes/malformed-attrs.stderr
+++ b/tests/ui/attributes/malformed-attrs.stderr
@@ -660,3 +660,69 @@ error: aborting due to 74 previous errors; 3 warnings emitted
 
 Some errors have detailed explanations: E0308, E0463, E0539, E0565, E0658, E0805.
 For more information about an error, try `rustc --explain E0308`.
+Future incompatibility report: Future breakage diagnostic:
+error: valid forms for the attribute are `#[doc(hidden|inline|...)]` and `#[doc = "string"]`
+  --> $DIR/malformed-attrs.rs:40:1
+   |
+LL | #[doc]
+   | ^^^^^^
+   |
+   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
+   = note: for more information, see issue #57571 <https://github.com/rust-lang/rust/issues/57571>
+   = note: `#[deny(ill_formed_attribute_input)]` on by default
+
+Future breakage diagnostic:
+error: valid forms for the attribute are `#[doc(hidden|inline|...)]` and `#[doc = "string"]`
+  --> $DIR/malformed-attrs.rs:73:1
+   |
+LL | #[doc]
+   | ^^^^^^
+   |
+   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
+   = note: for more information, see issue #57571 <https://github.com/rust-lang/rust/issues/57571>
+   = note: `#[deny(ill_formed_attribute_input)]` on by default
+
+Future breakage diagnostic:
+error: attribute must be of the form `#[link(name = "...", /*opt*/ kind = "dylib|static|...", /*opt*/ wasm_import_module = "...", /*opt*/ import_name_type = "decorated|noprefix|undecorated")]`
+  --> $DIR/malformed-attrs.rs:80:1
+   |
+LL | #[link]
+   | ^^^^^^^
+   |
+   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
+   = note: for more information, see issue #57571 <https://github.com/rust-lang/rust/issues/57571>
+   = note: `#[deny(ill_formed_attribute_input)]` on by default
+
+Future breakage diagnostic:
+error: valid forms for the attribute are `#[inline(always|never)]` and `#[inline]`
+  --> $DIR/malformed-attrs.rs:50:1
+   |
+LL | #[inline = 5]
+   | ^^^^^^^^^^^^^
+   |
+   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
+   = note: for more information, see issue #57571 <https://github.com/rust-lang/rust/issues/57571>
+   = note: `#[deny(ill_formed_attribute_input)]` on by default
+
+Future breakage diagnostic:
+error: valid forms for the attribute are `#[ignore = "reason"]` and `#[ignore]`
+  --> $DIR/malformed-attrs.rs:91:1
+   |
+LL | #[ignore()]
+   | ^^^^^^^^^^^
+   |
+   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
+   = note: for more information, see issue #57571 <https://github.com/rust-lang/rust/issues/57571>
+   = note: `#[deny(ill_formed_attribute_input)]` on by default
+
+Future breakage diagnostic:
+error: valid forms for the attribute are `#[ignore = "reason"]` and `#[ignore]`
+  --> $DIR/malformed-attrs.rs:220:1
+   |
+LL | #[ignore = 1]
+   | ^^^^^^^^^^^^^
+   |
+   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
+   = note: for more information, see issue #57571 <https://github.com/rust-lang/rust/issues/57571>
+   = note: `#[deny(ill_formed_attribute_input)]` on by default
+

--- a/tests/ui/feature-gates/issue-43106-gating-of-builtin-attrs-error.stderr
+++ b/tests/ui/feature-gates/issue-43106-gating-of-builtin-attrs-error.stderr
@@ -318,3 +318,14 @@ error: aborting due to 38 previous errors
 
 Some errors have detailed explanations: E0517, E0518, E0658.
 For more information about an error, try `rustc --explain E0517`.
+Future incompatibility report: Future breakage diagnostic:
+error: valid forms for the attribute are `#[inline(always|never)]` and `#[inline]`
+  --> $DIR/issue-43106-gating-of-builtin-attrs-error.rs:46:5
+   |
+LL |     #[inline = "2100"] fn f() { }
+   |     ^^^^^^^^^^^^^^^^^^
+   |
+   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
+   = note: for more information, see issue #57571 <https://github.com/rust-lang/rust/issues/57571>
+   = note: `#[deny(ill_formed_attribute_input)]` on by default
+

--- a/tests/ui/link-native-libs/link-attr-validation-early.stderr
+++ b/tests/ui/link-native-libs/link-attr-validation-early.stderr
@@ -19,3 +19,25 @@ LL | #[link = "foo"]
 
 error: aborting due to 2 previous errors
 
+Future incompatibility report: Future breakage diagnostic:
+error: attribute must be of the form `#[link(name = "...", /*opt*/ kind = "dylib|static|...", /*opt*/ wasm_import_module = "...", /*opt*/ import_name_type = "decorated|noprefix|undecorated")]`
+  --> $DIR/link-attr-validation-early.rs:2:1
+   |
+LL | #[link]
+   | ^^^^^^^
+   |
+   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
+   = note: for more information, see issue #57571 <https://github.com/rust-lang/rust/issues/57571>
+   = note: `#[deny(ill_formed_attribute_input)]` on by default
+
+Future breakage diagnostic:
+error: attribute must be of the form `#[link(name = "...", /*opt*/ kind = "dylib|static|...", /*opt*/ wasm_import_module = "...", /*opt*/ import_name_type = "decorated|noprefix|undecorated")]`
+  --> $DIR/link-attr-validation-early.rs:4:1
+   |
+LL | #[link = "foo"]
+   | ^^^^^^^^^^^^^^^
+   |
+   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
+   = note: for more information, see issue #57571 <https://github.com/rust-lang/rust/issues/57571>
+   = note: `#[deny(ill_formed_attribute_input)]` on by default
+

--- a/tests/ui/macros/issue-111749.stderr
+++ b/tests/ui/macros/issue-111749.stderr
@@ -16,3 +16,14 @@ LL |     cbor_map! { #[test(test)] 4};
 
 error: aborting due to 2 previous errors
 
+Future incompatibility report: Future breakage diagnostic:
+error: attribute must be of the form `#[test]`
+  --> $DIR/issue-111749.rs:8:17
+   |
+LL |     cbor_map! { #[test(test)] 4};
+   |                 ^^^^^^^^^^^^^
+   |
+   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
+   = note: for more information, see issue #57571 <https://github.com/rust-lang/rust/issues/57571>
+   = note: `#[deny(ill_formed_attribute_input)]` on by default
+

--- a/tests/ui/malformed/malformed-regressions.stderr
+++ b/tests/ui/malformed/malformed-regressions.stderr
@@ -46,3 +46,58 @@ LL | #[inline = ""]
 
 error: aborting due to 5 previous errors
 
+Future incompatibility report: Future breakage diagnostic:
+error: valid forms for the attribute are `#[doc(hidden|inline|...)]` and `#[doc = "string"]`
+  --> $DIR/malformed-regressions.rs:1:1
+   |
+LL | #[doc]
+   | ^^^^^^
+   |
+   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
+   = note: for more information, see issue #57571 <https://github.com/rust-lang/rust/issues/57571>
+   = note: `#[deny(ill_formed_attribute_input)]` on by default
+
+Future breakage diagnostic:
+error: attribute must be of the form `#[link(name = "...", /*opt*/ kind = "dylib|static|...", /*opt*/ wasm_import_module = "...", /*opt*/ import_name_type = "decorated|noprefix|undecorated")]`
+  --> $DIR/malformed-regressions.rs:7:1
+   |
+LL | #[link]
+   | ^^^^^^^
+   |
+   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
+   = note: for more information, see issue #57571 <https://github.com/rust-lang/rust/issues/57571>
+   = note: `#[deny(ill_formed_attribute_input)]` on by default
+
+Future breakage diagnostic:
+error: attribute must be of the form `#[link(name = "...", /*opt*/ kind = "dylib|static|...", /*opt*/ wasm_import_module = "...", /*opt*/ import_name_type = "decorated|noprefix|undecorated")]`
+  --> $DIR/malformed-regressions.rs:9:1
+   |
+LL | #[link = ""]
+   | ^^^^^^^^^^^^
+   |
+   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
+   = note: for more information, see issue #57571 <https://github.com/rust-lang/rust/issues/57571>
+   = note: `#[deny(ill_formed_attribute_input)]` on by default
+
+Future breakage diagnostic:
+error: valid forms for the attribute are `#[ignore = "reason"]` and `#[ignore]`
+  --> $DIR/malformed-regressions.rs:3:1
+   |
+LL | #[ignore()]
+   | ^^^^^^^^^^^
+   |
+   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
+   = note: for more information, see issue #57571 <https://github.com/rust-lang/rust/issues/57571>
+   = note: `#[deny(ill_formed_attribute_input)]` on by default
+
+Future breakage diagnostic:
+error: valid forms for the attribute are `#[inline(always|never)]` and `#[inline]`
+  --> $DIR/malformed-regressions.rs:5:1
+   |
+LL | #[inline = ""]
+   | ^^^^^^^^^^^^^^
+   |
+   = warning: this was previously accepted by the compiler but is being phased out; it will become a hard error in a future release!
+   = note: for more information, see issue #57571 <https://github.com/rust-lang/rust/issues/57571>
+   = note: `#[deny(ill_formed_attribute_input)]` on by default
+

